src/data/markdown/translated-guides/en/07 Misc/05 k6 Extensions.md

```diff
@@ -1,4 +1,197 @@
+```
+
+</CodeGroup>
+
+And confirm that `which xk6` on Linux/macOS or `where xk6.exe` on Windows returns a
+valid path. Otherwise ensure that `$GOPATH` is correctly defined and that
+`$GOPATH/bin` is added to your `$PATH` environment variable. See the
+[Go documentation](https://golang.org/cmd/go/#hdr-GOPATH_environment_variable) for details.
+
+Once xk6 is installed, building a k6 binary with one or more extensions can be done
+with the following command:
+
+<CodeGroup labels={[]} lineNumbers={[false]}>
+
+```bash
+$ xk6 build v0.33.0 \
+  --with github.com/imiric/xk6-sql \
+  --with github.com/szkiba/xk6-prometheus
+```
+
+</CodeGroup>
+
+This will build a `k6` binary in the current directory based on k6 v0.33.0, bundling
+a JavaScript and an Output extension. Now you can run a script with this binary that
+uses the [`xk6-sql` JS API](https://github.com/imiric/xk6-sql) and the
+[Prometheus output](https://github.com/szkiba/xk6-prometheus).
+
+Note that you'll need to specify the binary in the current directory (i.e. `./k6` or
+`.\k6` on Windows) to avoid using any other `k6` binary in your `PATH`.
+
+
+### Writing a new extension
+
+A good starting point for using xk6 and writing your own extension is the [xk6
+introductory article](https://k6.io/blog/extending-k6-with-xk6).
+<!-- TODO: How much of this article do we want to reproduce here? -->
+
+You'll first need to decide the type of extension you wish to create. Choose a
+JavaScript extension if you want to extend the JS functionality of your script, or an
+Output extension if you want to process the metrics emitted by k6 in some way.
+
+A few things to keep in mind when writing a new extension:
+<!-- TODO: Better structure this list? -->
+
+- JS extensions are registered using
+  [`modules.Register()`](https://pkg.go.dev/go.k6.io/k6@v0.33.0/js/modules#Register)
+  and Output extensions using
+  [`output.RegisterExtension()`](https://pkg.go.dev/go.k6.io/k6@v0.33.0/output#RegisterExtension),
+  which should be called in the `init()` function of the module.
+  In either case you'll have to specify a unique name that preferably doesn't clash
+  with any existing extension. JS extensions also must begin with `k6/x/` so as to
+  differentiate them from built-in JS modules.
+
+- JS extensions expose Go methods which can optionally receive a
+  [`context.Context`](https://golang.org/pkg/context/#Context) instance as the first argument.
+  This context is used throughout the k6 codebase and contains embedded objects
+  which can be extracted to access the
+  [`goja.Runtime`](https://pkg.go.dev/github.com/dop251/goja#Runtime) for a
+  particular VU, and to get more information about the test in progress.
+  Take a look at
+  [`common.GetRuntime()`](https://pkg.go.dev/go.k6.io/k6@v0.33.0/js/common#GetRuntime),
+  [`lib.GetState()`](https://pkg.go.dev/go.k6.io/k6@v0.33.0/lib#GetState) and
+  [`lib.GetExecutionState()`](https://pkg.go.dev/go.k6.io/k6@v0.33.0/lib#GetExecutionState).
```

Comment:
This might drastically change depending on the changes around https://github.com/k6io/k6/issues/1802, so maybe we should first decide on whether we will do any of the changes proposed there before we document stuff we might break

Reply:
Good point, but that will change in v0.34.0, right? Since we have versioned docs now, I think it's fine to mention the current API here.